### PR TITLE
add Lua examples

### PR DESCRIPTION
**Pull request by PedramNavid**
I really want some swag please dont make me learn fortran

### CODE REVIEW

File: lua/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Materialize + Lua Example
+
+This example uses [luasql](https://keplerproject.github.io/luasql/index.html)
+and was tested on Lua [5 4](5.4)
```
Comment (morsapaes) [inline lua/README.md:4]:
```suggestion
and was tested on Lua `v5.4`
```

File: lua/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+Install the postgres driver using luarocks:
```
Comment (morsapaes) [inline lua/README.md:6]:
```suggestion
Install the Postgres driver using LuaRocks:
```

File: lua/README.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+`luarocks install luasql-postgres`
+
+There are a few helper functions in `utils.lua` for iterating over rows
+and dumpings a table to stdout.
```
Comment (morsapaes) [inline lua/README.md:11]:
```suggestion
and dumping tables to `stdout`.
```

File: lua/insert.lua [hunk 4]
```diff
@@ -0,0 +1,28 @@
+local utils = require("utils")
+local driver = require "luasql.postgres"
+local env = assert (driver.postgres())
+local con = assert(env:connect("postgresql://materialize@localhost:6875/materialize?sslmode=disable"))
+
+assert (con:execute([[
+create table if not exists countries (name text, code text);
```
Comment (morsapaes) [inline lua/insert.lua:7]:
```suggestion
CREATE TABLE IF NOT EXISTS countries (name TEXT, code TEXT);
```

File: lua/query.lua [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+local con = assert(env:connect("postgresql://materialize@localhost:6875/materialize?sslmode=disable"))
+
+local function rows (connection, sql_statement)
+  local cursor = assert (connection:execute (sql_statement))
+  return function ()
+    return cursor:fetch()
+  end
+end
+
+for symbol, avg in rows(con, "select * from avg_bid") do
```
Comment (morsapaes) [inline lua/query.lua:12]:
```suggestion
for symbol, avg in rows(con, "SELECT * FROM avg_bid") do
```

File: lua/source.lua [hunk 6]
```diff
@@ -0,0 +1,20 @@
+local dump = require('utils').dump
+local driver = require "luasql.postgres"
+local env = assert (driver.postgres())
+local con = assert(env:connect("postgresql://materialize@localhost:6875/materialize?sslmode=disable"))
+
+con:execute[[
+    CREATE SOURCE market_orders_raw_2 from PUBNUB
```
Comment (morsapaes) [inline lua/source.lua:7]:
```suggestion
    CREATE SOURCE market_orders_raw_2 FROM PUBNUB
```

**Review by morsapaes — APPROVED** [review]
This is awesome, @PedramNavid! Left a few minor suggestions, mostly to make casing consistent. Thank you so much for contributing! ♥️

**Comment by PedramNavid** [comment]
> This is awesome, @PedramNavid! Left a few minor suggestions, mostly to make casing consistent. Thank you so much for contributing! hearts

thanks for reviewing! I accepted them all :]

**Pull request merged**